tests/commands/test-sdk.js

```diff
@@ -0,0 +1,802 @@
+
+					finished();
+				});
+			});
+
+			it('with pre-existing local sdk, mismatched timestamp, without --force', function (finished) {
+				const cli = {
+					argv: {
+						_: [ 'install' ],
+						$: 'titanium',
+						default: false,
+						force: false,
+						quiet: false,
+						'progress-bars': false,
+						version: path.join(__dirname, '../resources/mismatched_timestamp.zip')
+					},
+					env: {
+						installPath: sdksPath,
+						os: {
+							name: 'osx',
+							sdkPaths: [ sdksPath ],
+						},
+						sdks: {
+							'9.2.0.GA': {
+								manifest: {
+									name: '9.2.0.v20200923092031',
+									version: '9.2.0.GA',
+									timestamp: '9/23/2020 16:25'
+								},
+								path: path.join(sdksPath, 'mobilesdk/osx/9.2.0.GA')
+							}
+						}
+					},
+					addAnalyticsEvent: (name, obj) => {}
+				};
+
+				SDK.run(logger, config, cli, function (err) {
+					if (err) {
+						return finished(err);
+					}
+
+					logger.calls[0].should.eql([ 'banner', undefined ]);
+					logger.calls[1].should.eql([ 'log', `Extracting SDK to ${sdksPath.cyan}` ]);
+					logger.calls[2].should.eql([ 'log', `Titanium SDK ${'9.2.0'.cyan} successfully installed!\n` ]);
+
+					finished();
+				});
+			});
+
+			it('with bad sdk zipfile structure', function (finished) {
+				const cli = {
+					argv: {
+						_: [ 'install' ],
+						$: 'titanium',
+						default: false,
+						force: false,
+						quiet: false,
+						'progress-bars': false,
+						version: path.join(__dirname, '../resources/bad_folders.zip')
+					},
+					env: {
+						installPath: sdksPath,
+						os: {
+							name: 'osx',
+							sdkPaths: [ sdksPath ],
+						},
+						sdks: {
+							'9.2.0.GA': {
+								manifest: {
+									name: '9.2.0.v20200923092031',
+									version: '9.2.0.GA',
+									timestamp: '9/23/2020 16:25'
+								},
+								path: path.join(sdksPath, 'mobilesdk/osx/9.2.0.GA')
+							}
+						}
+					},
+					addAnalyticsEvent: (name, obj) => {}
+				};
+
+				SDK.run(logger, config, cli, function (err) {
+					try {
+						err.message.should.eql('Zip file did not contain expected manifest.json entry');
+					} catch (e) {
+						return finished(e);
+					}
+
+					finished();
+				});
+			});
+
+			it('with sdk zip missing manifest.json', function (finished) {
+				const cli = {
+					argv: {
+						_: [ 'install' ],
+						$: 'titanium',
+						default: false,
+						force: false,
+						quiet: false,
+						'progress-bars': false,
+						version: path.join(__dirname, '../resources/no_manifest.zip')
+					},
+					env: {
+						installPath: sdksPath,
+						os: {
+							name: 'osx',
+							sdkPaths: [ sdksPath ],
+						},
+						sdks: {
+							'9.2.0.GA': {
+								manifest: {
+									name: '9.2.0.v20200923092031',
+									version: '9.2.0.GA',
+									timestamp: '9/23/2020 16:25'
+								},
+								path: path.join(sdksPath, 'mobilesdk/osx/9.2.0.GA')
+							}
+						}
+					},
+					addAnalyticsEvent: (name, obj) => {}
```

Comment:
<!--
  0 failure: 
  2 warning:  tests/commands/te..., tests/commands/te...
  
  
  DangerID: danger-id-Danger;
  File: tests/commands/test-sdk.js;
  Line: 578;
-->

- :warning: tests/commands/test-sdk.js line 578 – 'name' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)
- :warning: tests/commands/test-sdk.js line 578 – 'obj' is defined but never used. Allowed unused args must match /^_.+/u. (no-unused-vars)